src/tarantool.client/ResponseReader.cs

```diff
@@ -29,14 +32,49 @@ internal class ResponseReader : IResponseReader
 
         private bool _disposed;
 
-        public ResponseReader(ILogicalConnection logicalConnection, ClientOptions clientOptions, INetworkStreamPhysicalConnection physicalConnection)
+        public ResponseReader(ClientOptions clientOptions, INetworkStreamPhysicalConnection physicalConnection)
         {
             _physicalConnection = physicalConnection;
-            _logicalConnection = logicalConnection;
             _clientOptions = clientOptions;
             _buffer = new byte[clientOptions.ConnectionOptions.ReadStreamBufferSize];
         }
 
+        public Task<MemoryStream> GetResponseTask(RequestId requestId)
+        {
+            lock (this)
```

Comment:
avoid concurrent access to _pendingRequests, while it perhaps may be destroying and should not accept requests any more